test/plugin/test_out_google_cloud.rb

```diff
@@ -176,6 +178,7 @@ def test_managed_vm_metadata_loading
   end
 
   def test_gce_metadata_does_not_load_when_fetch_gce_metadata_is_false
```

Comment:
My intent for introducing setup_gce_metadata_stubs was to allow us to act as a GCE environment (in which case we respond to the metadata server calls) or not (in which case the test framework will raise an exception).  I don't care if all the URLs are actually requested in the former case (we could check, but I think it would make the test more brittle than necessary), but we definitely care that they are *not* requested if we're not running on GCE which is how this test missed the bug.

But,  this test does have a bug; we shoudl not call setup_gce_metadata_stubs on the following line (181) since this test should succeed in a non-GCE environment.  That would have also caught the problem (I just verified by removing line 181 and also reverting the change to the plugin) independent of the new tests I added.